File: src_python/03_akar_kuadrat.py
# Program Mencari Akar Persamaan Kuadrat
a = float(input("Masukkan nilai a : "))
b = float(input("Masukkan nilai b : "))
c = float(input("Masukkan nilai c : "))

D = b**2 - 4*a*c

if D > 0:
    x1 = (-b + (D**0.5)) / (2*a)
    x2 = (-b - (D**0.5)) / (2*a)

    print("Akar real berbeda")
    print("x1 = %.3f" % x1)
    print("x2 = %.3f" % x2)

elif D == 0:
    x = -b / (2*a)

    print("Akar real kembar")
    print("x1 = x2 = %.3f" % x)

else:
    print("Persamaan memiliki akar-akar imajiner")

File: src_R/03_akar_kuadrat.R
# Program Mencari Akar Persamaan Kuadrat
a <- as.numeric(readline("Masukkan nilai a : "))
b <- as.numeric(readline("Masukkan nilai b : "))
c <- as.numeric(readline("Masukkan nilai c : "))

D <- b^2 - 4*a*c

if (D > 0) {
  
  x1 <- (-b + sqrt(D)) / (2*a)
  x2 <- (-b - sqrt(D)) / (2*a)
  
  cat("Akar real berbeda\n")
  cat(sprintf("x1 = %.3f\n", x1))
  cat(sprintf("x2 = %.3f\n", x2))
  
} else if (D == 0) {
  
  x <- -b / (2*a)
  
  cat("Akar real kembar\n")
  cat(sprintf("x1 = x2 = %.3f\n", x))
  
} else {
  
  cat("Persamaan memiliki akar-akar imajiner\n")
  
}
